ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditTextInputConnectionWrapper.java

```diff
@@ -94,10 +94,17 @@ public boolean setComposingText(CharSequence text, int newCursorPosition) {
     boolean noPreviousSelection = previousSelectionStart == previousSelectionEnd;
     boolean cursorDidNotMove = currentSelectionStart == previousSelectionStart;
     boolean cursorMovedBackwardsOrAtBeginningOfInput =
-        (currentSelectionStart < previousSelectionStart) || currentSelectionStart <= 0;
-    if (cursorMovedBackwardsOrAtBeginningOfInput || (!noPreviousSelection && cursorDidNotMove)) {
+      (currentSelectionStart < previousSelectionStart) || currentSelectionStart <= 0;
+    int maxLength = mEditText.getLengthFilterValue();
```

Comment:
Another way of getting the maxLength would be looping through the `InputFilters` on the `EditText` until we find the `InputLengthFilter`. Let me know if that approach is preferrable.